recipes/meson-python/ninja-overdep.patch

```diff
@@ -0,0 +1,32 @@
+diff --git a/PKG-INFO b/PKG-INFO
+index 8999cfd..6a86c0d 100644
+--- a/PKG-INFO
++++ b/PKG-INFO
+@@ -14,7 +14,6 @@ Project-URL: Changelog, https://mesonpy.readthedocs.io/en/latest/changelog.html
+ Requires-Python: >=3.7
+ Requires-Dist: colorama; os_name == "nt"
+ Requires-Dist: meson>=0.60.0
+-Requires-Dist: ninja
```

Comment:
This patch is kinda unusual, so let me summarize my understanding:
- `PKG-INFO` and `pyproject.toml` are not used at all by `conda` tooling, so no harm done there
- This is added to make `pip check` in CI happy

Maybe worth adding a comment if we're keeping this patch permanently?

Reply:
I'm not an expert in Python package metadata, but at least the `project.dependencies` list in `pyproject.toml` creates incorrect information on the requirements of this package, which will be present in the installed package and are therefore flagged by `pip`.

The `Requires-Dist:` tag and `build-system.requires` entries are probably less critical here, but are patched for consistency to make the intent.

Patching dependency metadata in projects from PyPI is not unusual IMO, at least for the projects I packaged so far I found it to be incorrect or incomplete in many cases.

Reply:
I can't say I understand this. `ninja` _is_ a required dependency. Beyond avoiding `pip check`, I'm not sure what this patch does or incorrect information means. Maybe one of the other reviewers knows?

Reply:
I guess the question boils down to whether `pip` is supposed to manage the dependency on a C++ program or not.

Reply:
Ah, I guess we just came to the same conclusion of that being the question:)  https://github.com/FFY00/meson-python/issues/60#issuecomment-1133957875

Reply:
Just as an fyi mostly: Pip check is not necessary. It’s nice to have but fails in many situations where a dependency doesn‘t properly declare itself (like here). 
I‘d be happy to merge the PR without it if that's easier for you.

Reply:
I would prefer to keep this patch to have tools like `build`, `pip` or `wheel` which work with the package metadata not flag the unrecognized dependency from the conda package. Of course it is possible to manually silence those warnings and checks, but this moves the burden to the user.

Reply:
0.5.0 should be released later today with some changes to dependencies, may be worth waiting for that and updating this tomorrow.

Reply:
Updated to 0.5.0